connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -175,11 +201,13 @@ public boolean startConnector(
             final ConnectorConfig connConfig = new ConnectorConfig(connProps);
             final String connClass = connConfig.getString(ConnectorConfig.CONNECTOR_CLASS_CONFIG);
             log.info("Creating connector {} of type {}", connName, connClass);
-            final Connector connector = connectorFactory.newConnector(connClass);
+            final Connector connector = plugins.newConnector(connClass);
             workerConnector = new WorkerConnector(connName, connector, ctx, statusListener);
             log.info("Instantiated connector {} with version {} of type {}", connName, connector.version(), connector.getClass());
+            ClassLoader save = plugins.compareAndSwapLoaders(connector);
```

Comment:
Yes, classloader swapping needs to be reviewed overall. Right now it has been placed on a "per request" fashion while running the connectors. 

Here I just fixed swapping back when an Exception occurs.